### PR TITLE
[ADD] add module 'web_widget_color_tree_field'

### PR DESCRIPTION
**Pull request by damdam-s**
allow to change background and text color for fields in tree view according to data's value in the record being rendered.

### CODE REVIEW

**Comment by gurneyalex** [comment]
:+1:

File: web_widget_color_tree_field/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (bealdav) [inline web_widget_color_tree_field/__init__.py]: Please save the cloud of these bytes ;-)

**Comment by bealdav** [comment]
Yeah, like in my dreams, thanks a lot.
LGTM, no test
:+1:

File: web_widget_color_tree_field/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    'author': "Camptocamp,Odoo Community Association (OCA)",
+    'license': 'AGPL-3',
+    'website': 'http://www.camptocamp.com',
+    'data': [
+        'views/web_widget_color_tree_field.xml',
+    ],
+    'qweb': [
+        'static/xml/*.xml',
+    ],
+    'auto_install': False
```
Comment (legalsylvain) [inline web_widget_color_tree_field/__openerp__.py]: Hi, this line is unecessary.

**Comment by mdietrichc2c** [comment]
LGTM, :+1:

File: web_widget_color_tree_field/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'web_widget_color_tree_field',
```
Comment (legalsylvain) [inline web_widget_color_tree_field/__openerp__.py]: Please, remove underscore in the friendly name.

File: web_widget_color_tree_field/README.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+* Add attribute 'bg_color' to color background of a cell in tree view
+
+* Add attribute 'fg_color' to change text color of a cell in tree view
+
+
+Usage
+=====
+
+* In the tree view declaration, put bg_color="red:customer==True;" attribute in the field tag::
+
```
Comment (legalsylvain) [inline web_widget_color_tree_field/README.rst]: I tried on runbot, setting 
<field name="phone" bg_color="red:customer==True;"/> in the partner tree view, and it doesn't work. Is this module working when we change via UI ? 
Or did I missed something ?
Kind regards.

Comment (damdam-s) [inline web_widget_color_tree_field/README.rst]: @legalsylvain 
I forgot to mention that :
- after installing : click on the enterprise/Odoo logo to reload javascript files
- fields mentionned in attributes must be present in tree view (even in invisible). This is the same behaviour as colors in tree tag.

**Comment by pedrobaeza** [comment]
The name for the module is not correct, as this is not a widget, but a modification to tree view. A more proper name I think is _web_tree_dynamic_coloured_field_

**Comment by gurneyalex** [comment]
let's use the US spelling then: color and not colour.

**Comment by pedrobaeza** [comment]
Colored is then something reasonably to put: https://en.wikipedia.org/wiki/Colored?

**Comment by nhomar** [comment]
> > The name for the module is not correct, as this is not a widget, but a modification to tree view. A more proper name I think is web_tree_dynamic_coloured_field

Agreed with pedro, falling appart of this. :+1:

**Comment by damdam-s** [comment]
rename done !

**Comment by pedrobaeza** [comment]
Tried on runbot and working as expected.

:+1:

**Pull request merged**